### PR TITLE
Remove OpenHW/SiLabs copyright headers from unmodified Coremark files

### LINKED ISSUE #1623: Clean-up of Coremark directories for license clarity
### Task Outcome
Currently, Coremark benchmark is integrated on a per core basis. In addition, the original sources seem to have been modified. In consequence, a Silicon Labs/OpenHW group copyright has been added to the header, in addition to a solderpad license. The header does not explain what the changes are that have been done to the files. 
The issue is twofold:
1. It is not clear what was modified (i.e. what is subject to the respective copyright
2. Expectation would be that the benchmark files are not modified to ensure the integrity of the benchmark

My expectation would be that the coremark files are taken unmodified, straight from the eembc github. Best case, you don't even copy the files but just pull them when building. The respective core directories in the repository should only contain the core_portme{.c/.h} files, of which the .h file should most likely also not be modified.

Provide a brief description of what is expected.  For example, "Implement an
XYZ UVM Agent." If necessary point to a specification document.

### Background information
https://github.com/eembc/coremark

### Location Information
core-v-verif/*core*/tests/programs/custom/coremark/

### Completion Criteria
Clear copyright and license situation. Original Coremark files included.

### PR DESCRIPTION
**Pull request by MikeOpenHWGroup**
This PR is the first step to resolve #1623.

It updates all of the Coremark license headers in `core-v-verif/<core>/tests/programs/custom/coremark` to remove the OpenHW/SiLabs copyright from unmodified files.

@christian-herber-nxp, if/when you sign off on this I will make the same changes to the `cv32e20` branch.

### CODE REVIEW

File: cv32e40p/tests/programs/custom/coremark/test.yaml [hunk 1]
```diff
@@ -1,4 +1,21 @@
+# Copyright 2020 OpenHW Group
+# Copyright 2020 Silicon Labs, Inc.
+# SPDX-License-Identifier: Apache-2.0 WITH SHL-2.1
 name: coremark
 uvm_test: uvmt_cv32e40p_firmware_test_c
+default_cflags: >
```
Comment (christian-herber-nxp) [inline cv32e40p/tests/programs/custom/coremark/test.yaml:6]: looks like this is unrelated to the current issue, right?

File: cv32e40p/tests/programs/custom/coremark/test.yaml [hunk 2]
```diff
@@ -1,4 +1,21 @@
+# Copyright 2020 OpenHW Group
```
Comment (christian-herber-nxp) [inline cv32e40p/tests/programs/custom/coremark/test.yaml:1]: # Copyright 2020, 2023 OpenHW Group

Note: I would see no copyright statements on such simple yaml files as ok, as the content doesn't have a sufficient creation level. But if you put them, put them right

**Comment by christian-herber-nxp** [comment]
i will approve it, as it fixes the critical things. Updating the copyright year would be appreciated

File: cv32e40p/tests/programs/custom/coremark/test.yaml (hunk 1 above)
Comment (MikeOpenHWGroup) [reply cv32e40p/tests/programs/custom/coremark/test.yaml:6]: Correct.  This yaml did not have `ITERATIONS` defined, so the test-program did not compile.  I solved the issue by copying the `default_cflags` from the CV32E40S version of this test.  The two cores are sufficiently similar that I _believe_ this is valid.

File: cv32e40p/tests/programs/custom/coremark/test.yaml (hunk 2 above)
Comment (christian-herber-nxp) [reply cv32e40p/tests/programs/custom/coremark/test.yaml:1]: i dont follow. the header in the yaml file state 2020 only. as you modified it in 2023, i would expect this to be reflected

**Comment by MikeOpenHWGroup** [comment]
Thanks for the review @christian-herber-nxp.  If you are satified with my responses, please mark the conversations as resolved.

@silabs-robin, @YoannPruvost, do either of you have any comments?

File: cv32e40p/tests/programs/custom/coremark/test.yaml (hunk 1 above)
Comment (christian-herber-nxp) [reply cv32e40p/tests/programs/custom/coremark/test.yaml:6]: there is a default behavior in coremark, where it automatically determines the number of needed iterations. i think coremark requires 10k iterations or so for a valid result, which is not practical in simulation. anyway, performance has very low variance in such low end CPUs, so the result after 10 iterations is already pretty accurate. For a CVA6, this could already be different

**Review by YoannPruvost — APPROVED** [review]

**Review by christian-herber-nxp — APPROVED** [review]

**Comment by silabs-robin** [comment]
> [...] do either of you have any comments?

For files that truly had no modifications from the originals, I don't expect that it should be any problem to remove the added copyright header.

**Comment by MikeOpenHWGroup** [comment]
Thanks for the reviews, comments and approvals @YoannPruvost, @silabs-robin and @christian-herber-nxp. Sadly, none of you are Contributors, so I cannot yet merge it in. @silabs-hfegran, can you have a look?   Thanks!

**Comment by DBees** [comment]
@MikeOpenHWGroup (for clarification) I think you mean none are committers

**Comment by MikeOpenHWGroup** [comment]
Yes, @DBees.  I meant to say Committers.
